generators/auto-update/AutoUpdateUtils.js

```diff
@@ -0,0 +1,84 @@
+    static async autoUpdate(type, localVersion, generatorVersion, destinationPath, task) {
+        const repo = path.basename(destinationPath);
+        const excecuteUpdates = AutoUpdateUtils.getAvailableUpdates().filter((version) => semver.gtr(version, localVersion));
+        return task.newListr(
+            [...excecuteUpdates.map((version) => {
+                return {
+                    title: 'update ' + version,
+                    options: {
+                        bottomBar: Infinity,
+                        persistentOutput: true
+                    },
+                    task: async (ctx, task) => ctx[repo].descriptions.push(await AutoUpdateUtils.updateLogic(version, generatorVersion, type, destinationPath, task, ctx))
+                };
+            })], { exitOnError: true, concurrent: false, rendererOptions: { showErrorMessage: true, collapseErrors: false, collapse: false } }
+        );
+    }
+
+    static async updateLogic(nextVersion, generatorVersion, type, destinationPath, task, ctx) {
+        const filePath = `./updates/update-${nextVersion}.js`;
+        const repo = path.basename(destinationPath);
+        const { update, description } = require(filePath);
+        const currentVersion = AutoUpdateUtils.readConfig('localVersion', destinationPath) || NpmUtils.decrementVersion(generatorVersion);
+        const setCtx = (key, value) => {
+            ctx[repo][key] = value;
+        };
+        if (currentVersion === nextVersion) {
+            throw new Error(`${repo}: Duplicate version tag "${currentVersion}"`);
+        }
+        return update(type, destinationPath, task, setCtx)
+            .then(async () => {
+                if (ctx[repo].skip) {
+                    return;
+                }
+                AutoUpdateUtils.setConfig('localVersion', nextVersion, destinationPath);
+                return `#### ${currentVersion} to ${nextVersion}\n ${description}`;
+            })
+            .catch((e) => {
+                const msg = `${repo}: Update ${currentVersion} to ${nextVersion} failed with ${e.message}`;
+                e.message = msg;
+                // does it make sense if you call the revert function? 
+                throw e;
+            });
+    }
+
+    static getAvailableUpdates() {
+        const files = glob('update-*.js', {
+            cwd: path.join(__dirname, 'updates')
+        }) || [];
+        const versions = files.map((file) => file.match(/(?<=update-).*?(?=.js)/)[0]);
+        return semver.sort(versions);
+    }
+
+    static setConfig(key, value, cwd) {
+        const target = path.join(cwd + '/.yo-rc.json');
+        const file = fse.readJSONSync(target);
+        file['generator-phovea'][key] = value;
+        fse.writeJSONSync(target, file, { spaces: 2 });
+    }
+
+    static readConfig(key, cwd) {
+        const file = fse.readJSONSync(path.join(cwd + '/.yo-rc.json'));
+        return file['generator-phovea'][key];
+    }
+
+    static chooseCredentials(org) {
```

Comment:
I think we should think there more globally - so f.e. we can say: username: process.env.{org}_USER and token: process.env.{org}_TOKEN